File: apprefurbished/src/main/java/com/androidcollider/easyfin/main/MainMVP.java
package com.androidcollider.easyfin.main;

import androidx.annotation.Nullable;

import io.reactivex.Flowable;

/**
 * @author Ihor Bilous
 */

public interface MainMVP {

    interface Model {

        Flowable<Integer> getAccountsCountObservable();
    }

    interface View {

        void informNoAccounts();
    }

    interface Presenter {

        void setView(@Nullable MainMVP.View view);

        void checkIsAccountsExists();
    }
}

File: apprefurbished/src/main/java/com/androidcollider/easyfin/main/MainMVP.kt
package com.androidcollider.easyfin.main

import io.reactivex.rxjava3.core.Single

/**
 * @author Ihor Bilous
 */
interface MainMVP {
    interface Model {
        val accountsCountObservable: Single<Int>?
    }

    interface View {
        fun informNoAccounts()
    }

    interface Presenter {
        fun setView(view: View?)
        fun checkIsAccountsExists()
    }
}